src/WordPress/Blueprints/BlueprintMapper.php

```diff
@@ -4,16 +4,22 @@
 
 
 use WordPress\Blueprints\Model\DataClass\Blueprint;
+use WordPress\Blueprints\Model\DataClass\ModelInfo;
 use WordPress\JsonMapper\JsonMapper;
+use WordPress\JsonMapper\JsonMapperException;
 
 class BlueprintMapper {
-	protected $mapper;
+	private $mapper;
+
+	private $custom_factories = array();
 
 	public function __construct() {
 		$this->mapper = new JsonMapper();
+		$this->prepare_factory_for_resources();
+		$this->prepare_factory_for_steps();
+		$this->mapper->configure_evaluators( $this->custom_factories );
```

Comment:
Nitpick, how about an API like this?

```suggestion
		$this->mapper = new JsonMapper(array(
			$this->create_resource_mapper(),
			$this->create_steps_mapper()
		));

```

Reply:
Done.